app/models/shipit/stack.rb

```diff
@@ -502,7 +502,12 @@ def schedule_merges_if_necessary
 
     def emit_lock_hooks
       return unless previous_changes.include?('lock_reason')
-      Hook.emit(:lock, self, locked: locked?, stack: self)
+
+      lock_duration = if previous_changes['lock_reason'].last.blank?
+        {from: previous_changes['locked_since'].first, until: Time.zone.now}
```

Comment:
The value of this doesn't match the name. Could we emit an integer here (`lock_duration_in_seconds`)? I think that would make the testing easier too (e.g. probably wouldn't need to add `Timecop`).

Reply:
`_in_seconds` is a bit verbose for no reason, so I'd say `lock_duration`, but 👍

Reply:
I wouldn't want to change that as we would loose the start and the end time of the lock. It is better to safe the raw data and do the calculation later on.

Reply:
nit: `lock_details` or similar would have been better in that case, the value of this field is not a duration.

Reply:
makes sense. I'll update that tomorrow.